lib/connectors/memory.js

```diff
@@ -237,6 +242,34 @@ Memory.prototype.updateOrCreate = function (model, data, options, callback) {
   });
 };
 
+Memory.prototype.findOrCreate = function(model, filter, data, callback) {
+  var self = this;
+  var nodes = self._findAllSkippingIncludes(model, filter);
+
+  if (nodes.length > 0) {
+    if (filter && filter.include) {
+      self._models[model].model.include(nodes[0], filter.include, {}, function(err, nodes) {
+        process.nextTick(function() {
+          if (err) return callback(err);
+          callback(null, nodes[0], false);
+        });
+      });
+    } else {
+      process.nextTick(function() {
+        callback(null, nodes[0], false);
+      });
+    }
+  } else {
+    this._create(model, data, function(err, id) {
+      if (err) return callback(err);
+      self.saveToFile(id, function(err, id) {
+        self.setIdValue(model, data, id);
+        callback(err, data, true);
+      });
+    });
+  }
+};
```

Comment:
Nitpick, feel free to ignore: long if/else branches are difficult to read, using an early returns (possibly with inverse logic in `if`) is usually better. In this case, I would personally structure the code as follows - notice how the last block has 2-levels of indentation less:

```js
Memory.prototype.findOrCreate = function(model, filter, data, callback) {
  var self = this;
  var nodes = self._findAllSkippingIncludes(model, filter);
  var found = nodes[0];

  if (!found) {
    return this._create(model, data, function(err, id) {
      if (err) return callback(err);
      self.saveToFile(id, function(err, id) {
        self.setIdValue(model, data, id);
        callback(err, data, true);
      });
    });
  }

  if (!filter || !filter.include) {
    return process.nextTick(function() {
      callback(null, found, false);
    });
  }

  self._models[model].model.include(found, filter.include, {}, function(err, nodes) {
    process.nextTick(function() {
      if (err) return callback(err);
      callback(null, found, false);
    });
  });
};
```